app/models/user.rb

```diff
@@ -18,6 +22,21 @@ class User < ApplicationRecord
   # TODO: samuel - ActiveStorage validations needs to be discussed and implemented.
   validate :avatar_validation
 
+  def room_owner?(room)
+    id == room.user_id
+  end
+
+  def room_shared?(room)
+    shared_rooms.pluck(:friendly_id).include?(room.friendly_id)
+  end
```

Comment:
I wouldn't put this in the User model - I'd probably put it in the Room model instead

Reply:
They are not needed anymore I will remove them